src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,43 @@
 package core.basesyntax;
 
-class Pair {
+import java.util.Objects;
 
+class Pair<T, S> {
+    private T first;
+    private S second;
+
+    private Pair(T first, S second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public static <T, S> Pair of(T first, S second) {
+        return new Pair(first, second);
+    }
+
+    public T getFirst() {
+        return first;
+    }
+
+    public S getSecond() {
+        return second;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Pair<?, ?> pair = (Pair<?, ?>) o;
+        return Objects.equals(first, pair.first)
```

Comment:
try to do it without Objects class and same hashCode()
Hint: check yourself with checklist before sending your solution

Reply:
Done. Thank you!